File: apps/agent/src/lib/briefings.ts
import { ObjectId, type Collection } from "mongodb";
import { getDb } from "./mongo.js";

export interface BriefingRecord {
  _id?: ObjectId;
  eventId: ObjectId;
  eventTitle: string;
  eventWhen: string | null;
  eventLocation: string | null;
  attendees: string[];
  markdown: string;
  contextSummary?: string | null;
  citations?: Array<{
    chunkId: string;
    documentId: string;
    source: string;
    title: string;
    score: number;
    ordinal: number;
  }>;
  model?: string;
  createdAt: Date;
}

export async function briefingsCol(): Promise<Collection<BriefingRecord>> {
  const db = await getDb();
  return db.collection<BriefingRecord>("briefings");
}

export async function saveBriefing(rec: Omit<BriefingRecord, "_id">): Promise<string> {
  const col = await briefingsCol();
  const res = await col.insertOne({ ...rec });
  return res.insertedId.toString();
}

export async function getBriefing(id: string): Promise<BriefingRecord | null> {
  if (!ObjectId.isValid(id)) return null;
  const col = await briefingsCol();
  return col.findOne({ _id: new ObjectId(id) });
}

export async function listBriefings(limit = 30): Promise<BriefingRecord[]> {
  const col = await briefingsCol();
  return col
    .find({}, { limit, sort: { createdAt: -1 } })
    .toArray();
}

export async function findBriefingByEvent(eventId: string): Promise<BriefingRecord | null> {
  if (!ObjectId.isValid(eventId)) return null;
  const col = await briefingsCol();
  return col.findOne(
    { eventId: new ObjectId(eventId) },
    { sort: { createdAt: -1 } },
  );
}

export function publicBriefing(b: BriefingRecord): Record<string, unknown> {
  return {
    id: b._id?.toString() ?? "",
    eventId: b.eventId.toString(),
    eventTitle: b.eventTitle,
    eventWhen: b.eventWhen,
    eventLocation: b.eventLocation,
    attendees: b.attendees,
    markdown: b.markdown,
    contextSummary: b.contextSummary ?? null,
    citations: b.citations ?? [],
    model: b.model ?? null,
    createdAt: b.createdAt instanceof Date ? b.createdAt.toISOString() : b.createdAt,
  };
}

File: apps/agent-py/app/lib/briefings.py
"""Briefings store — port of apps/agent/src/lib/briefings.ts."""
from __future__ import annotations

from datetime import datetime

from bson import ObjectId

from app.db.mongo import collection


def briefings_col():
    return collection("briefings")


async def save_briefing(rec: dict) -> str:
    ins = await briefings_col().insert_one({**rec})
    return str(ins.inserted_id)


async def get_briefing(bid: str) -> dict | None:
    if not ObjectId.is_valid(bid):
        return None
    return await briefings_col().find_one({"_id": ObjectId(bid)})


async def list_briefings(limit: int = 30) -> list[dict]:
    return await briefings_col().find({}, limit=limit, sort=[("createdAt", -1)]).to_list(length=None)


def _iso(v):
    return v.isoformat() if isinstance(v, datetime) else v


def public_briefing(b: dict) -> dict:
    return {
        "id": str(b.get("_id", "")), "eventId": str(b.get("eventId", "")),
        "eventTitle": b.get("eventTitle"), "eventWhen": b.get("eventWhen"),
        "eventLocation": b.get("eventLocation"), "attendees": b.get("attendees", []),
        "markdown": b.get("markdown", ""), "contextSummary": b.get("contextSummary"),
        "citations": b.get("citations", []), "model": b.get("model"),
        "createdAt": _iso(b.get("createdAt")),
    }
